### ISSUE TITLE
--fix broken for arrow-parens and trailing-comma together

### ISSUE THREAD

**Issue by ismail-syed** [palantir/tslint#2877]
### Bug Report

We're looking at integrating [prettier](https://github.com/prettier/prettier/releases/tag/1.4.0) with our tslint config setup. With Prettier's v1.4.0 release, Typescript formatting is now supported!

However, prettiers auto-formatting can conflict with tslint config, so the suggested workflow is to run your ts code through prettier, then run tslint fixes. I tested this our in our project and noticed some obscure tslint autofixing using `--fix` in CLI. This only occurs when the code is formatted as below with `arrow-parens` and `trailing-comma` is enabled.

From the following code, I am aware that `.filter` can be simplified into one line. However, in the actual source where this snippet is coming from `key === 'foo'` is a very long line that should ideally be in its own line.

This issue has come up a number of times in our codebase when running prettier + tslint --fix. I'd be interested in fixing this, with some guidance. I considered reproducing this with a testcase but couldn't find test cases for --fix behavior IIRC.

- __TSLint version__: 5.4.2
- __TypeScript version__: 2.3.2
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts

Object.keys({'foo': 1, 'bar': 2, 'baz': 3})
  .filter(
    key =>
      key === 'foo'
  )
  .map(key => console.log(key));
```

with `tslint.json` configuration:

```json

"rules" : {
 "arrow-parens": true,
  "trailing-comma": [true, { "multiline": "always", "singleline": "never" }]
}

```

#### Actual behavior
<img width="331" alt="screen shot 2017-06-03 at 11 57 38 pm" src="https://cloud.githubusercontent.com/assets/6130700/26758828/99181e6e-48b8-11e7-89d5-8d0e211335aa.png">

#### Expected behavior
<img width="334" alt="screen shot 2017-06-03 at 11 58 20 pm" src="https://cloud.githubusercontent.com/assets/6130700/26758826/91bd2f42-48b8-11e7-8ceb-799b4858668f.png">

**Comment by ajafff** [comment]
I guess you are using the `--project` CLI argument? This issue looks very similar to #2862.

That should be fixed by #2864

**Comment by ismail-syed** [comment]
Yes.
